dask/array.py

```diff
@@ -373,6 +373,16 @@ def _slice_1d(dim_shape, lengths, index):
 
     >>> _slice_1d(100, [20, 20, 20, 20, 20], 25)
     {1: 5}
+
+    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 0, -3))
+    {0: slice(-2, -20, -3), 1: slice(-1, -21, -3), 2: slice(-3, -21, -3), 3: slice(-2, -21, -3), 4: slice(-1, -21, -3)}
+
+    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 12, -3))
+    {0: slice(-2, -8, -3), 1: slice(-1, -21, -3), 2: slice(-3, -21, -3), 3: slice(-2, -21, -3), 4: slice(-1, -21, -3)}
+
+    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, -12, -3))
+    {4: slice(-1, -12, -3)}
```

Comment:
Cool.  I actually had to verify these using an interpreter.  Good to know that we're beyond what's intuitive.